src/main/scala/com/activegrid/model/Graph.scala

```diff
@@ -0,0 +1,21 @@
+package com.activegrid.model
+
+import eu.fakod.neo4jscala.{EmbeddedGraphDatabaseServiceProvider, Neo4jWrapper}
+import org.neo4j.graphdb.Node
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+/**
+  * Created by shareefn on 30/9/16.
+  */
+
+object Graph  {
+
+  trait Neo4jRep[T] {
+
+    def toNeo4jGraph(entity: T): Node
+
+    def fromNeo4jGraph(id: Option[Long]): Option[T]
```

Comment:
For entities within entities, if inner entity is optional while persisting, we get may or may not get child id while converting it to case class.

So I am passing this Optional id directly to fromNeo4jGraph, which will take care of handling this situation